### PR TITLE
[INLONG-8674][Manager] Pulsar - Modify the calling method from SDK to HTTP

### LINKED ISSUE #8674: [Improve][Manager] Pulsar - Modify the calling method from SDK to HTTP
### Description

The available API could refer to https://pulsar.apache.org/admin-rest-api/#.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haibo-duan**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8674

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

Modify the pulsar calling method from SDK to HTTP.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Comment by dockerzhang** [comment]
@haibo-duan please rebase from the master to solve the conflict, thanks.

**Comment by dockerzhang** [comment]
Do we need to remove the pulsar dependency in the pom for this PR?

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/util/HttpUtils.java [hunk 1]
```diff
@@ -112,6 +112,22 @@ public static <T> T request(RestTemplate restTemplate, String url, HttpMethod ht
         return response.getBody();
     }
 
+    /**
+     * Send an void HTTP request
+     */
+    public static void request(RestTemplate restTemplate, String url, HttpMethod httpMethod, Object requestBody,
+            HttpHeaders header) {
+        if (log.isDebugEnabled()) {
+            log.debug("begin request to {} by request body {}", url, GSON.toJson(requestBody));
```
Comment (vernedeng) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/util/HttpUtils.java:121]: just print the debug log directly

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java [hunk 2]
```diff
@@ -17,18 +17,29 @@
...
-import org.apache.inlong.manager.common.util.Preconditions;
+import org.apache.inlong.manager.common.util.HttpUtils;
 import org.apache.inlong.manager.pojo.cluster.pulsar.PulsarClusterInfo;
+import org.apache.inlong.manager.pojo.queue.pulsar.PulsarLookupTopicInfo;
+import org.apache.inlong.manager.pojo.queue.pulsar.PulsarNamespacePolicies;
+import org.apache.inlong.manager.pojo.queue.pulsar.PulsarPartitionedInternalStats;
+import org.apache.inlong.manager.pojo.queue.pulsar.PulsarTenantInfo;
+import org.apache.inlong.manager.pojo.queue.pulsar.PulsarTopicMetadata;
 
+import com.alibaba.fastjson.JSON;
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java:28]: please use Jackson

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/queue/pulsar/PulsarNamespacePolicies.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class PulsarNamespacePolicies {
+
+    private int message_ttl_in_seconds;
```
Comment (vernedeng) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/queue/pulsar/PulsarNamespacePolicies.java:31]: camel style

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java [hunk 4]
```diff
@@ -260,31 +264,42 @@ public void createSubscription(PulsarAdmin pulsarAdmin, String fullTopicName, St
...
-            this.createSubscription(pulsarAdmin, fullTopicName, topicInfo.getQueueModule(), subscription);
+            this.createSubscription(pulsarClusterInfo, fullTopicName, topicInfo.getQueueModule(), subscription);
         }
         LOGGER.info("success to create subscription={} for multiple topics={}", subscription, topicList);
     }
 
     /**
      * Check if Pulsar tenant exists
+     *
+     * @param pulsarClusterInfo
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java:280]: Please add a specific description, like:
```
     /**
     * Poll the Sort status infos by the given inlong groups
     *
     * @param groupInfos inlong group infos
     * @param credentials credential info
     * @return list of Sort status infos
     * @throws Exception any exception if occurred
     */
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java [hunk 5]
```diff
@@ -260,31 +264,42 @@ public void createSubscription(PulsarAdmin pulsarAdmin, String fullTopicName, St
...
 
     /**
-     * Check whether the Pulsar namespace exists under the specified tenant
+     * Check whether the Pulsar namespace exists under the specified tenant.
+     *
+     * @param pulsarClusterInfo
+     * @param tenant
+     * @param namespace
+     * @return
+     * @throws Exception
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java:297]: Ditto.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java [hunk 6]
```diff
@@ -39,54 +51,420 @@ public class PulsarUtils {
...
+    public static final String QUERY_PERSISTENT_PATH = "/admin/v2/persistent";
+    public static final String LOOKUP_TOPIC_PATH = "/lookup/v2/topic";
+
+    private static final Gson GSON = new GsonBuilder().create(); // thread safe
+
+    /**
+     * get http headers by token.
+     *
+     * @param token
+     * @return
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java:67]: Ditto.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java [hunk 7]
```diff
@@ -39,54 +51,420 @@ public class PulsarUtils {
...
+    }
+
     /**
-     * Get pulsar admin info
+     * Get pulsar cluster info list.
+     *
+     * @param restTemplate
+     * @param clusterInfo
+     * @return
+     * @throws Exception
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java:83]: Ditto.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java [hunk 8]
```diff
@@ -39,54 +51,420 @@ public class PulsarUtils {
...
+                ArrayList.class);
+    }
+
+    /**
+     * Get the list of active brokers.
+     *
+     * @param restTemplate
+     * @param clusterInfo
+     * @return
+     * @throws Exception
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java:98]: Ditto.

**Review by fuweng11 — APPROVED** [review]

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/queue/pulsar/PulsarPartitionedInternalStats.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Map;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class PulsarPartitionedInternalStats {
```
Comment (aloyszhang) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/queue/pulsar/PulsarPartitionedInternalStats.java:31]: The real partitioned-stats contains more field.
Can the query result(from a real pulsar cluster) be deserialized correctly?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java [hunk 10]
```diff
@@ -39,54 +51,420 @@ public class PulsarUtils {
...
+     * Get the list of active brokers.
+     *
+     * @param restTemplate spring framework RestTemplate
+     * @param clusterInfo pulsar cluster info
+     * @return list of pulsar broker infos
+     * @throws Exception any exception if occurred
+     */
+    public static List<String> getPulsarBrokers(RestTemplate restTemplate, PulsarClusterInfo clusterInfo)
+            throws Exception {
+        final String url = clusterInfo.getAdminUrl() + QUERY_BROKERS_PATH;
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java:102]: url should contains cluster name, like `http://localhost:8080/admin/v2/brokers/$clusterName`

File: inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/queue/PulsarUtilsTest.java [hunk 11]
```diff
@@ -17,39 +17,161 @@
...
+    @BeforeAll
+    public static void before() {
+        PULSAR_CLUSTER_INFO.setAdminUrl(DEFAULT_SERVICE_URL);
+        restTemplate = new RestTemplate();
+    }
+
+    /**
+     * Just using in local test.
+     */
+    @Disabled
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/queue/PulsarUtilsTest.java:66]: why all tests are disabled?

**Review by aloyszhang — CHANGES_REQUESTED** [review]
Plz add tests to truly cover this change.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/queue/pulsar/PulsarPartitionedInternalStats.java (hunk 9 above)
Comment (haibo-duan) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/queue/pulsar/PulsarPartitionedInternalStats.java]: modify to jsonObject

File: inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/queue/PulsarUtilsTest.java (hunk 11 above)
Comment (haibo-duan) [reply inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/queue/PulsarUtilsTest.java]: Added test case again.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java [hunk 12]
```diff
@@ -39,54 +65,919 @@ public class PulsarUtils {
...
+     * @param restTemplate spring framework RestTemplate
+     * @param clusterInfo pulsar cluster info
+     * @return list of pulsar broker infos
+     * @throws Exception any exception if occurred
+     */
+    public static List<String> getPulsarBrokers(RestTemplate restTemplate, PulsarClusterInfo clusterInfo)
+            throws Exception {
+        List<String> clusters = getPulsarClusters(restTemplate, clusterInfo);
+        List<String> brokers = new ArrayList<>();
+        for (String brokerName : brokers) {
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java:121]: why traverse an empty list?

**Review by aloyszhang — COMMENTED** [review]
Overall LGTM, left some comments, PTAL

File: inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/queue/PulsarUtilsTest.java [hunk 13]
```diff
@@ -17,39 +17,671 @@
...
+     * Test cases for {@link PulsarUtils#getPulsarClusters}.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testGetPulsarClusters() throws Exception {
+        final String url = DEFAULT_SERVICE_URL + PulsarUtils.QUERY_CLUSTERS_PATH;
+        final String result = "[\"standalone\"]";
+        List<String> expected = GSON.fromJson(result, ArrayList.class);
+        when(restTemplate.exchange(eq(url), eq(HttpMethod.GET), any(HttpEntity.class), eq(String.class))).thenReturn(
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/queue/PulsarUtilsTest.java:143]: It's better to run a pulsar broker service in the test, and then use the pulsar service to verify all these HTTP requests work well.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java [hunk 14]
```diff
@@ -39,54 +65,919 @@ public class PulsarUtils {
...
+
+    /**
+     * Get pulsar cluster info list.
+     *
+     * @param restTemplate spring framework RestTemplate
+     * @param clusterInfo pulsar cluster info
+     * @return list of pulsar cluster infos
+     * @throws Exception any exception if occurred
+     */
+    public static List<String> getPulsarClusters(RestTemplate restTemplate, PulsarClusterInfo clusterInfo)
```
[the remark this answers is not in the document]
Comment (haibo-duan) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java:102]: OK

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java [hunk 15]
```diff
@@ -273,7 +273,7 @@ public static void createPartitionedTopic(RestTemplate restTemplate, PulsarClust
      * @return pulsar internal stat info of partitioned topic
      * @throws Exception any exception if occurred
      */
-    public static JsonObject getPulsarStatsPartitionedTopics(RestTemplate restTemplate,
+    public static JsonObject getStatsPartitionedTopics(RestTemplate restTemplate,
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java:276]: First of all, `partitioned-stats` and `partitioned-statsInternal` are two different REST APIs for the pulsar.
The method name `getStatsPartitionedTopics` is not match with the URL `clusterInfo.getAdminUrl() + QUERY_PERSISTENT_PATH + "/" + topicPath + "/partitioned-internalStats"`

**Review by aloyszhang — DISMISSED** [review]
Nice contribution!

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java [hunk 16]
```diff
@@ -39,54 +66,921 @@ public class PulsarUtils {
...
+     * @return
      */
-    public static String getServiceUrl(PulsarAdmin pulsarAdmin, String pulsarCluster) throws PulsarAdminException {
-        return pulsarAdmin.clusters().getCluster(pulsarCluster).getServiceUrl();
+    private static int readVarInt(ByteBuffer buf) {
+        byte tmp = buf.get();
+        if (tmp >= 0) {
+            return tmp;
+        } else {
+            int result = tmp & 127;
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java:858]: Can we avoid using so many if-else here.

Comment (haibo-duan) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java:858]: These codes are copied from readVarInt method of org.apache.pulsar.common.api.proto.LightProtoCodec class，I think it's best to be consistent with the original codes.

**Review by aloyszhang — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by haifxu — APPROVED** [review]

**Pull request merged**